clippy_lints/src/methods/mod.rs

```diff
@@ -1498,11 +1498,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Methods {
                 if cx.access_levels.is_exported(impl_item.hir_id) {
                 // check missing trait implementations
                     for &(method_name, n_args, fn_header, self_kind, out_type, trait_name) in &TRAIT_METHODS {
+                        let no_lifetime_params = || {
+                            impl_item.generics.params.iter().filter(|p| match p.kind {
+                               hir::GenericParamKind::Lifetime { .. } => true,
+                               _ => false,
+                            }).count() == 0
+                        };
                         if name == method_name &&
                             sig.decl.inputs.len() == n_args &&
                             out_type.matches(cx, &sig.decl.output) &&
                             self_kind.matches(cx, self_ty, first_arg_ty) &&
-                            fn_header_equals(*fn_header, sig.header) {
+                            fn_header_equals(*fn_header, sig.header) &&
+                            // ignore methods with lifetime params, risk of false positive
```

Comment:
I'm a little worried about blanket restricting methods with lifetimes in them but I cant find any reason why this would cause problems or any traits that would get a false negative as a result of this change so this seems fine.